### PR TITLE
Resource/learningResource

### LINKED ISSUE #1: Learning resource
```
{
   category: 'coding' || 'maths' || 'electronics',
   ages: [number, number],
   title: string,
   rating: {
     type: number,
     max: 5,
     min: 0
  },
  body: string,
  tags: {
    type: array,
    of: strings
  }
}
```

### PR DESCRIPTION
**Pull request by DaveDangereux**
Closes #1

### CODE REVIEW

File: src/resources/learningResource/learningResource.model.js
Comment (shanesc) [inline src/resources/learningResource/learningResource.model.js]: Do we need to require ages? What about 'All ages' cases or no max age?

File: src/resources/learningResource/learningResource.model.js
Comment (shanesc) [inline src/resources/learningResource/learningResource.model.js:24]: Might make sense to require a description in the body. What do you think?

File: src/resources/learningResource/learningResource.model.js
Comment (shanesc) [inline src/resources/learningResource/learningResource.model.js]: I'd say tags should be optional since we have the required category already, what do you think?

**Review by shanesc — COMMENTED** [review]
Looks really good Dave!

I added a few comments on the model, not required changes but just wanted to see what you thought before we merge. Let me know.

File: src/resources/learningResource/learningResource.model.js
Comment (r1903) [reply src/resources/learningResource/learningResource.model.js:24]: I think as description field is optional in the front end form.

File: src/resources/learningResource/learningResource.model.js
Comment (r1903) [reply src/resources/learningResource/learningResource.model.js]: In the resource form(from figma) age field is mandatory, so it should be fine.

File: src/resources/learningResource/learningResource.model.js
Comment (DaveDangereux) [reply src/resources/learningResource/learningResource.model.js]: This was an oversight, actually. The required flag only prevents me from setting the field to null - if I omit the field on the request, an empty array is added instead.

I've removed it for now, as I agree that they should be optional anyway.

File: src/resources/learningResource/learningResource.model.js
Comment (DaveDangereux) [reply src/resources/learningResource/learningResource.model.js:24]: It definitely feels like requiring a description would be a good idea here. I was imagining somebody might want to draft a card and add details later, though perhaps this is not intended functionality.

I see that the description field is optional, so for parity with the design I think I'll leave this for now.

File: src/resources/learningResource/learningResource.model.js
Comment (DaveDangereux) [reply src/resources/learningResource/learningResource.model.js]: Hmm... I obviously hadn't looked properly at the design. The 'age' field is a single drop-down menu. If these were just a set of hard-coded options in the form, perhaps it would make more sense for this field to be a single string rather than two numbers. That might also allow the simple inclusion of 'all ages' as an option, but it's potentially a bit limiting as well.

Happy to cross that bridge when we come to it, though.

It would be amusing to attend an event with a minimum age of 3 but no maximum age...

Comment (DaveDangereux) [reply src/resources/learningResource/learningResource.model.js]: Changed my mind.

I think based on the design it makes sense for the field to be a string, but mandatory. Then the options can be specified in the form. Shall edit.

Comment (shanesc) [reply src/resources/learningResource/learningResource.model.js]: Ah yeah makes sense, sorry I didn't actually check the design in Figma! 🤦‍♂️

Yeah we can update it later if need be 👍

File: src/resources/learningResource/learningResource.model.js
Comment (shanesc) [reply src/resources/learningResource/learningResource.model.js:24]: Yep, agreed keep it as per the design.

File: src/resources/learningResource/learningResource.model.js
Comment (shanesc) [reply src/resources/learningResource/learningResource.model.js]: Oh that's interesting... so if the type is an array and required, an empty value passed in will still pass validation, and an empty array put in?

Comment (shanesc) [reply src/resources/learningResource/learningResource.model.js]: Just noticed, I think your most recent commit overwrote the deletion of the `required: true` property

Comment (shanesc) [reply src/resources/learningResource/learningResource.model.js]: Gonna approve anyway, go ahead and change if needed or Squash & Merge if you're happy with it 👍

**Review by shanesc — APPROVED** [review]
All good! Remove the required field from the `tags` property if needed, otherwise go ahead and hit "Squash and Merge"!

File: src/resources/learningResource/learningResource.model.js
Comment (DaveDangereux) [reply src/resources/learningResource/learningResource.model.js]: Thanks for spotting that! Not sure how that crept in again...

**Pull request merged**
